### ISSUE TITLE
cache_from when ECR registry doesn't exist causes panic

### ISSUE THREAD

**Issue by ShedPlant** [pulumi/pulumi-docker#576]
### What happened?

Trying to deploy a new Pulumi project for the first time.
It defines an AWS ECR Repository, and an `Image`.
The `Image` has `cache_from` set to the ECR repository explicitly. As far as I know, this is necessary if I want the 2nd+ build to be fast.

The ECR Repo does not yet exist. I am _guessing_ this is relevant because I have seen `pulumi-docker` work on existing projects.

`pulumi preview` or `pulumi up` causes a panic!

```
    panic: interface conversion: interface {} is resource.Computed, not string
    goroutine 101 [running]:
    github.com/pulumi/pulumi/sdk/v3/go/common/resource.PropertyValue.StringValue(...)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.56.0/go/common/resource/properties.go:419
    github.com/pulumi/pulumi-docker/provider/v4.marshalCachedImages({{0x211cf40?, 0xc000420cf0?}})
    	/home/runner/work/pulumi-docker/pulumi-docker/provider/image.go:414 +0x22b
    github.com/pulumi/pulumi-docker/provider/v4.(*dockerNativeProvider).Check(0xc000578b80, {0x24070e8, 0xc000420c90}, 0xc0004222a0)
    	/home/runner/work/pulumi-docker/pulumi-docker/provider/provider.go:194 +0x88d
    github.com/pulumi/pulumi-docker/provider/v4.dockerHybridProvider.Check({{}, {0x2dd8d80, 0x25dfe, 0x25dfe}, {0x23f231c, 0x6}, {0x24136b0, 0xc00022b180}, {0x2413600, 0xc000578b80}}, ...)
    	/home/runner/work/pulumi-docker/pulumi-docker/provider/hybrid.go:96 +0xd7
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Check_Handler.func1({0x24070e8, 0xc000420c90}, {0x20db800?, 0xc0004222a0})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.56.0/proto/go/provider_grpc.pb.go:537 +0x78
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x24070e8, 0xc000420930}, {0x20db800, 0xc0004222a0}, 0xc000876180, 0xc000126690)
    	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Check_Handler({0x2152be0?, 0xc000790320}, {0x24070e8, 0xc000420930}, 0xc00074e0e0, 0xc000577380)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.56.0/proto/go/provider_grpc.pb.go:539 +0x138
    google.golang.org/grpc.(*Server).processUnaryRPC(0xc000276000, {0x240f360, 0xc00050d040}, 0xc00037eea0, 0xc000792030, 0x2e1a6f0, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0xd33
    google.golang.org/grpc.(*Server).handleStream(0xc000276000, {0x240f360, 0xc00050d040}, 0xc00037eea0, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0xa36
    google.golang.org/grpc.(*Server).serveStreams.func1.2()
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0x98
    created by google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

```

If I comment out the `cache_from` line, the panic does not occur.

### Expected Behavior

ECR Repository to be created
No caching can happen on first time
Image to be built
Image to be pushed

### Steps to reproduce

`my_app/deployment/__main__.py`:
```python
import base64
from pathlib import Path

import pulumi
import pulumi_aws as aws
import pulumi_docker as docker

PROJECT_DIR = Path(__file__).resolve().parent.parent


# https://github.com/pulumi/pulumi-docker/blob/master/examples/aws-container-registry/py/__main__.py
# Get registry info (creds and endpoint) so we can build/publish to it.
def _get_registry_info(rid):
    creds = aws.ecr.get_credentials(registry_id=rid)
    decoded = base64.b64decode(creds.authorization_token).decode()
    parts = decoded.split(":")
    if len(parts) != 2:
        raise Exception("Invalid credentials")
    return docker.RegistryArgs(
        server=creds.proxy_endpoint,
        username=parts[0],
        password=parts[1],
    )


ecr_repo = aws.ecr.Repository(
    resource_name="my_app:repo",
    name="my_app",
)

registry = ecr_repo.registry_id.apply(_get_registry_info)

image = docker.Image(
    "my_app:image",
    build=docker.DockerBuildArgs(
        context=str(PROJECT_DIR),
        dockerfile=str(PROJECT_DIR / "Dockerfile"),
        platform="linux/amd64",
        target="prod",
        # BREAKS START
        cache_from=docker.CacheFromArgs(images=[pulumi.Output.concat(ecr_repo.repository_url, ":latest")])
        # BREAKS END
    ),
    registry=registry,
    image_name=pulumi.Output.concat(ecr_repo.repository_url, "my_app:latest"),
    skip_push=True,
)
```

`Dockerfile`:
```Dockerfile
FROM python:3.9-slim-bullseye as base

RUN echo "The Dockerfile contents are irrelevant"
```

### Output of `pulumi about`

```
CLI          
Version      3.60.0
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.33.0
docker      4.1.1

Host     
OS       darwin
Version  13.2.1
Arch     x86_64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by ShedPlant** [comment]
FYI @DAtek

**Comment by ShedPlant** [comment]
https://github.com/curoo/pulumi-docker-investigations/tree/master/issues/issue_576

https://github.com/curoo/pulumi-docker-investigations/actions/runs/4576462157/jobs/8080636229

**Comment by guineveresaenger** [comment]
hi @ShedPlant - thank you for helping us out by filing this issue. I pushed a [fix](#577) and it should go out in the next patch.
